File: 1-Patron/php/GestorDeArmadura.php
<?php
class GestorDeArmadura
{
  private $armadura;

  function __construct($armadura)
  {
    $this->armadura = $armadura;
  }

  function mejorarArmadura($mejora)
  {
    $this->armadura->setNivel($mejora);
  }

  function repararArmadura($reparacion)
  {
    $this->armadura->incDuracion($reparacion);
  }

  function generarArmadura($nombre, $nivel, $dano_recibido, $duracion)
  {
    return new Armadura($nombre, $nivel, $dano_recibido, $duracion);
  }
}
?>

File: 1-Patron/Java/GestordeArmadura.java
package masterwholeproxy;

public class GestordeArmadura{
    private Armadura armadura;
    
    public GestordeArmadura(Armadura armadura){
        this.armadura = armadura;
    }
    public boolean mejorarArmadura(int mejora){
        this.armadura.setNivel(mejora);
        return true;
    }
    public boolean repararArmadura(int incremento){
        this.armadura.incDuracion(incremento);
        return true;
    }
    public Armadura generarArmadura(String nombre,int nivel,float dano,float duracion){
        return (new Armadura(nombre,nivel,dano,duracion));
    }
    
}   
